components/playlist/playlist_service.cc

```diff
@@ -242,6 +257,171 @@ void PlaylistService::ConfigureWebPrefsForBackgroundWebContents(
+  }
+
+  std::move(callback).Run(std::move(items));
+}
+
+void PlaylistService::GetPlaylistItem(const std::string& id,
+                                      GetPlaylistItemCallback callback) {
+  DCHECK(!id.empty());
+  const auto* item_value = prefs_->GetDict(kPlaylistItemsPref).FindDict(id);
+  DCHECK(item_value);
+  if (!item_value) {
+    std::move(callback).Run({});
+    return;
+  }
+
+  return std::move(callback).Run(GetPlaylistItemFromValue(*item_value));
+}
+
+void PlaylistService::AddMediaFilesFromPageToPlaylist(
+    const std::string& playlist_id,
+    const GURL& url) {
+  VLOG(2) << __func__ << " " << playlist_id << " " << url;
+  PlaylistDownloadRequestManager::Request request;
+  request.url_or_contents = url.spec();
+  request.callback = base::BindOnce(
+      &PlaylistService::AddMediaFilesFromItems, weak_factory_.GetWeakPtr(),
+      playlist_id.empty() ? GetDefaultSaveTargetListID() : playlist_id,
+      prefs_->GetBoolean(playlist::kPlaylistCacheByDefault));
+  download_request_manager_->GetMediaFilesFromPage(std::move(request));
+}
+
+void PlaylistService::AddMediaFilesFromTabToPlaylist(
+    int32_t window_id,
+    int32_t tab_id,
+    const std::string& playlist_id) {
+  DCHECK(delegate_);
+
+  auto* contents = delegate_->GetWebContents(window_id, tab_id);
+  if (!contents)
+    return;
+
+  AddMediaFilesFromContentsToPlaylist(
+      playlist_id, contents, prefs_->GetBoolean(kPlaylistCacheByDefault));
+}
+
+void PlaylistService::FindMediaFilesFromTab(
+    int32_t window_id,
+    int32_t tab_id,
+    FindMediaFilesFromTabCallback callback) {
+  DCHECK(delegate_);
+
+  auto* contents = delegate_->GetWebContents(window_id, tab_id);
+  if (!contents)
+    return;
+
+  PlaylistDownloadRequestManager::Request request;
+  request.url_or_contents = contents->GetWeakPtr();
+  request.callback = std::move(callback);
+  download_request_manager_->GetMediaFilesFromPage(std::move(request));
+}
+
+void PlaylistService::AddMediaFiles(std::vector<mojom::PlaylistItemPtr> items,
+                                    const std::string& playlist_id) {
+  AddMediaFilesFromItems(playlist_id,
+                         prefs_->GetBoolean(playlist::kPlaylistCacheByDefault),
+                         std::move(items));
+}
+
+void PlaylistService::RemoveItemFromPlaylist(const std::string& playlist_id,
+                                             const std::string& item_id) {
+  RemoveItemFromPlaylist(PlaylistId(playlist_id), PlaylistItemId(item_id));
+}
+
+void PlaylistService::MoveItem(const std::string& from_playlist_id,
+                               const std::string& to_playlist_id,
+                               const std::string& item_id) {
+  MoveItem(PlaylistId(from_playlist_id), PlaylistId(to_playlist_id),
+           PlaylistItemId(item_id));
+}
+
+void PlaylistService::UpdateItem(mojom::PlaylistItemPtr item) {
+  UpdatePlaylistItemValue(item->id,
+                          base::Value(GetValueFromPlaylistItem(item)));
```

Comment:
why are we converting between `PlaylistItem` and base::Value? You can pass base::Value directly over mojo